apps/src/sites/studio/pages/scripts/show.js

```diff
@@ -97,6 +97,9 @@ function initPage() {
   // rendered on this page
   updateQueryParam('completedLessonNumber', undefined);
 
+  const unitHasLevels =
+    scriptData.lessons.reduce((n, {levels}) => n + levels.length, 0) > 0;
```

Comment:
This way
a) Computes a boolean the whole way through, and
b) Still works even if `levels` is `null`/`undefined`/some object without a `length` property.

```suggestion
  const unitHasLevels =
    scriptData.lessons.reduce((n, {levels}) => n || !!levels?.length, false);
```

Reply:
nice update! I can't remember the last time I wrote a boolean reducer